### PR TITLE
Implemented Control: "Disable creation of classic pipeline"

### PR DESCRIPTION
**Pull request by Anjali-1012**
## Description
</br>
Disable the creation of classic build and classic release pipelines.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 1]
```diff
@@ -3847,4 +3847,37 @@ class Organization: ADOSVTBase
...
+       {
+            $orgLevelScope = $this.PipelineSettingsObj.disableClassicPipelineCreation #the name of the getter setter
+
+            if($orgLevelScope -eq $true )
+            {
+                $controlResult.AddMessage([VerificationResult]::Passed, "Creation of classic build and classic release pipelines is disabled.");
+            }
+            else{
+                $controlResult.AddMessage([VerificationResult]::Failed, "Creation of classic build and classic release pipelines is not disabled.");
+                if ($this.ControlFixBackupRequired -or $this.BaselineConfigurationRequired)
```
Comment (saikumarra) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:3863]: Why do we need control backup here?

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Organization.json [hunk 2]
```diff
@@ -672,6 +672,25 @@
         "AutomatedFix"
       ],
       "Enabled": true
-  }
+    },
+    {
+      "ControlID": "ADO_Organization_DP_Disable_Creation_Of_Classic_Pipeline",
+      "Description": "Disable the creation of classic build and classic release pipelines.",
+      "Id": "Organization500",
+      "ControlSeverity": "Medium",
```
Comment (saikumarra) [inline src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Organization.json:680]: How we decided the severity?

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Organization.json [hunk 3]
```diff
@@ -672,6 +672,25 @@
...
+      "Automated": "Yes",
+      "MethodName": "CheckDisableCreationOfClassicPipelines",
+      "Rationale": "Classic pipelines have been deprecated. YAML pipelines provide better pipeline security with their ability to allow developers to review pipeline code, better resource access management through approvals and checks and support for runtime parameters to avoid multiple security issues. Disabling creation of classic pipelines ensure that developers get to work with most secure options in the environment without worrying about security of their classic pipelines.",
+      "Recommendation": "1.Go to Organization Settings --> 2.Pipelines --> 3.Settings --> 4.Enable 'Disable creation of classic build and classic release pipelines.",
```
Comment (saikumarra) [inline src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Organization.json:684]: Can we also implement Autofix for this?

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Organization.json (hunk 2 above)
Comment (Anjali-1012) [reply src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Organization.json:680]: The same control is implemented in AzTS framework and there the severity is medium.
So I kept it as medium.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 (hunk 1 above)
Comment (Anjali-1012) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1]: Sorry, I added it by mistake. I removed it now.

**Review by saikumarra — DISMISSED** [review]

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 4]
```diff
@@ -3847,4 +3847,25 @@ class Organization: ADOSVTBase
         }
         return $controlResult
     }
+    hidden [ControlResult] CheckDisableCreationOfClassicPipelines([ControlResult] $controlResult)
+    {
+       $controlResult.VerificationResult = [VerificationResult]::Failed
+       if($this.PipelineSettingsObj)
+       {
+            $orgLevelScope = $this.PipelineSettingsObj.disableClassicPipelineCreation #the name of the getter setter
```
Comment (rbavdekar) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:3855]: Please use a better variable to define $orgLevelScope

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 [hunk 5]
```diff
@@ -4207,4 +4207,38 @@ class Project: ADOSVTBase
         }
         return $controlResult;
     }
+
+    hidden [ControlResult] CheckDisabledCreationOfClassicPipeline([ControlResult] $controlResult)
+    {
+        $controlResult.VerificationResult = [VerificationResult]::Failed
+        if($this.PipelineSettingsObj)
+        {
+            $orgLevelScope = $this.PipelineSettingsObj.disableClassicPipelineCreation.orgEnabled;
```
Comment (rbavdekar) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4216]: Is $orgLevelScope check required? If the setting is enabled at org level, it is automatically enabled at project level and hence you will get true in $projLevelScope itself. Second check may be unnecessary. Please validate if the orgLevelScope use case flow will ever happen

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 [hunk 6]
```diff
@@ -4207,4 +4207,38 @@ class Project: ADOSVTBase
...
+            if($prjLevelScope -eq $true )
+            {
+                $controlResult.AddMessage([VerificationResult]::Passed, "Creation of classic build and classic release pipelines is disabled.");
+            }
+            else
+            {
+                $controlResult.AddMessage([VerificationResult]::Failed, "Creation of classic build and classic release pipelines is not disabled.");
+            }
+
+            if($orgLevelScope -eq $true )
```
Comment (rbavdekar) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4229]: If else are not written correctly. If prjLevelScope is true our control should pass. But if code is written in this manner, VerificationResult will always come in accordance of orgLevelScope. 

Case prjLevelScope= true and orgLevelScope=false.
VerificationResult will be Failed even though it should be passed

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 (hunk 5 above)
Comment (Anjali-1012) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4216]: Yes, If the setting is enabled at org level, it is automatically enabled at project level. I referred the code of a similar control. For that control it is checked for both, org level and project level. So I too checked for both.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 (hunk 6 above)
Comment (Anjali-1012) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4229]: While I was debugging the code for this control both the messages were seen. That is the project level result and org level result.
Will check it again.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 (hunk 5 above)
Comment (rbavdekar) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4216]: Which control? Let me check if there is an API issue with Juhi

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 (hunk 6 above)
Comment (rbavdekar) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4229]: Both messages will be seen but verificationResult populated will be wrong

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 (hunk 5 above)
Comment (rbavdekar) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4216]: You can keep only project. Remove code for checking org level

Comment (Anjali-1012) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:4216]: okay

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 (hunk 6 above)
Comment (Anjali-1012) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1]: okay, removed the org level check.

**Review by juhi037 — APPROVED** [review]

**Review by rbavdekar — APPROVED** [review]

**Review by juhi037 — APPROVED** [review]

**Pull request merged**